File: src/main/java/net/maple/handlers/user/UserPortalScriptRequestHandler.java
package net.maple.handlers.user;

import client.Client;
import field.object.portal.FieldPortal;
import net.maple.handlers.PacketHandler;
import util.packet.PacketReader;

public class UserPortalScriptRequestHandler extends PacketHandler {

    @Override
    public void handlePacket(PacketReader reader, Client c) {
        reader.readByte();

        String name = reader.readMapleString();
        FieldPortal portal = c.getCharacter().getField().getPortalByName(name);

        if (portal != null) {
            if (!portal.getScript().isEmpty()) {
                System.out.println(portal);
                c.getCharacter().enableActions();
            }
        }
    }
}

File: src/main/java/net/maple/handlers/user/UserPortalScriptRequestHandler.kt
package net.maple.handlers.user

import client.Client
import net.maple.handlers.PacketHandler
import util.logging.LogType
import util.logging.Logger
import util.packet.PacketReader

class UserPortalScriptRequestHandler : PacketHandler {

    override fun handlePacket(reader: PacketReader, c: Client) {
        val chr = c.character
        val fieldKey = reader.readByte()
        if (fieldKey != chr.fieldKey) {
            return run {
                Logger.log(LogType.FIELD_KEY, "FieldKey mismatch", this, c)
                c.close(this, "FieldKey mismatch")
            }
        }

        val name = reader.readMapleString()

        val portal = chr.field.getPortalByName(name) ?: return

        if (portal.script.isNotEmpty()) {
            println(portal)
            chr.enableActions()
        }
    }
}